### PR TITLE
feat: agilent tapestation adapter

### PR DESCRIPTION
**Pull request by slopez-b**
Add Agilent TapeStation Analysis adapter

### CODE REVIEW

File: src/allotropy/allotrope/schemas/adm/electrophoresis/BENCHLING/2024/06/electrophoresis.schema.json [hunk 1]
```diff
@@ -113,7 +113,7 @@
                                       "$asm.pattern": "quantity datum",
                                       "allOf": [
                                         {
-                                          "$ref": "http://purl.allotrope.org/json-schemas/adm/core/REC/2024/03/core.schema#/$defs/tQuantityValue"
+                                          "$ref": "#/$defs/tQuantityValue"
```
Comment (nathan-stender) [inline src/allotropy/allotrope/schemas/adm/electrophoresis/BENCHLING/2024/06/electrophoresis.schema.json:116]: Why this change?

File: src/allotropy/parsers/agilent_tapestation_analysis/agilent_tapestation_analysis_parser.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    def _get_electrophoresis_document(
+        self, metadata: MetaData, samples: list[Sample]
+    ) -> list[ElectrophoresisDocumentItem]:
+        return [
+            ElectrophoresisDocumentItem(
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_document=[
+                        MeasurementDocumentItem(
+                            measurement_identifier=sample.measurement_identifier,
+                            measurement_time=sample.measurement_time,
```
Comment (nathan-stender) [inline src/allotropy/parsers/agilent_tapestation_analysis/agilent_tapestation_analysis_parser.py:113]:
```suggestion
                            measurement_time=self._get_date_time(sample.measurement_time),
```

File: src/allotropy/parsers/agilent_tapestation_analysis/agilent_tapestation_analysis_parser.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+                        value=peak.relative_corrected_peak_area
+                    ),
+                    comment=peak.comment,
+                )
+                for peak in peaks
+            ]
+        )
+
+    def _get_data_region_document(
+        self, data_regions: list[DataRegion], unit_cls: PEAK_UNIT_CLASSES
```
Comment (nathan-stender) [inline src/allotropy/parsers/agilent_tapestation_analysis/agilent_tapestation_analysis_parser.py:198]: Nit: either make this `peak_unit_cls` or make `peak_unit_cls` above `unit_cls`

File: src/allotropy/parsers/agilent_tapestation_analysis/agilent_tapestation_analysis_structure.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+                    DataSource(feature=feature, reference=Referenceable(uuid=source_id))
+                ],
+            )
+        )
+
+    return calculated_data
+
+
+@dataclass(frozen=True)
+class MetaData:
```
Comment (nathan-stender) [inline src/allotropy/parsers/agilent_tapestation_analysis/agilent_tapestation_analysis_structure.py:56]: Nit: rename to `Metadata`

**Review by nathan-stender — CHANGES_REQUESTED** [review]
Code looks great!

One question and some small style requests

File: src/allotropy/allotrope/schemas/adm/electrophoresis/BENCHLING/2024/06/electrophoresis.schema.json (hunk 1 above)
Comment (slopez-b) [reply src/allotropy/allotrope/schemas/adm/electrophoresis/BENCHLING/2024/06/electrophoresis.schema.json:116]: The `tQuantityValue` defined in the allotrope published `REC/2024/03/core.schema` does only support numeric values. Since we need to also support non numeric values ("NaN"), this change is necessary for the schema validation to pass.

Comment (nathan-stender) [reply src/allotropy/allotrope/schemas/adm/electrophoresis/BENCHLING/2024/06/electrophoresis.schema.json:116]: Ah gotcha 👍

**Review by nathan-stender — APPROVED** [review]
